build/vfs.js

```diff
@@ -55,30 +75,38 @@ export default function env(inst) {
         buffer_ptr,
         amount,
       );
-      Deno.seekSync(rid, offset, Deno.SeekMode.Start);
-      return Deno.writeSync(rid, buffer);
+      const file = getOpenFile(rid);
+      file.seekSync(offset, Deno.SeekMode.Start);
+      return file.writeSync(buffer);
     },
     // Truncate the given file
     js_truncate: (rid, size) => {
-      Deno.ftruncateSync(rid, size);
+      const file = getOpenFile(rid);
+      file.truncateSync(size);
     },
     // Sync file data to disk
     js_sync: (rid) => {
-      Deno.fdatasyncSync(rid);
+      const file = getOpenFile(rid);
+      file.syncDataSync();
     },
     // Retrieve the size of the given file
     js_size: (rid) => {
-      return Deno.fstatSync(rid).size;
+      const file = getOpenFile(rid);
+      return file.statSync().size;
     },
     // Acquire a SHARED or EXCLUSIVE file lock
     js_lock: (rid, exclusive) => {
       // this is unstable and has issues on Windows ...
-      if (Deno.flockSync && !isWindows) Deno.flockSync(rid, exclusive !== 0);
+      if (Deno.flockSync && !isWindows) {
```

Comment:
@dyedgreen What do you think is the best way to detect the availability of `FsFile.lockSync`?

* `FsFile.lockSync` is available in v1.44.0, but `Deno.flockSync` is still undefined without `--unstable` or `--unstable-fs`. So we can't check `Deno.flockSync`.
* `FsFile.lockSync` exists in versions before v1.44.0, but calling it terminates deno, even from inside of a try/catch block. So we can't check the existence of `FsFile.lockSync`.
* `Deno.version.deno >= "1.44.0"` is the only condition I could get to work. Should we go with that?

Reply:
Yeah I feel like we’ll just need to do a version check here 😕